ActiveQuery.php

```diff
@@ -352,7 +352,7 @@ protected function executeScript($db, $type, $columnName = null)
         // find by primary key if possible. This is much faster than scanning all records
         if (is_array($this->where) && (
                 !isset($this->where[0]) && $modelClass::isPrimaryKey(array_keys($this->where)) ||
-                isset($this->where[0]) && $this->where[0] === 'in' && $modelClass::isPrimaryKey($this->where[1])
+                isset($this->where[0]) && $this->where[0] === 'in' && $modelClass::isPrimaryKey((array)$this->where[1])
```

Comment:
You missed a space between `(array)` and casting value. Even within this file the other occurences of `(array)` are followed by space, so this is inconsistent and small code style violation.